assert/assertions.go

```diff
@@ -429,14 +429,27 @@ func samePointers(first, second interface{}) bool {
 // to a type conversion in the Go grammar.
 func formatUnequalValues(expected, actual interface{}) (e string, a string) {
 	if reflect.TypeOf(expected) != reflect.TypeOf(actual) {
-		return fmt.Sprintf("%T(%#v)", expected, expected),
-			fmt.Sprintf("%T(%#v)", actual, actual)
+		return fmt.Sprintf("%T(%s)", expected, truncatingFormat(expected)),
+			fmt.Sprintf("%T(%s)", actual, truncatingFormat(actual))
 	}
 	switch expected.(type) {
 	case time.Duration:
 		return fmt.Sprintf("%v", expected), fmt.Sprintf("%v", actual)
 	}
-	return fmt.Sprintf("%#v", expected), fmt.Sprintf("%#v", actual)
+	return truncatingFormat(expected), truncatingFormat(actual)
+}
+
+// The output of the error message is eventually read a line at time by a bufio.Scanner
+// which has a max line length of bufio.MaxScanTokenSize, if the value we are displaying is too long
+// then the error message is not shown.  So lets trucate realy long values so we can
+// at least see some of the error message.
+func truncatingFormat(expected interface{}) string {
```

Comment:
Could you please add some unit tests for this function?

Reply:
yep